### PR TITLE
Fix link warnings on macOS

### PR DESCRIPTION
**Pull request by battlmonstr**
Fixes https://github.com/torquem-ch/silkworm/issues/674 and https://github.com/torquem-ch/silkworm/issues/673 and https://github.com/torquem-ch/silkworm/issues/688

References:
https://github.com/protocolbuffers/protobuf/issues/8958
https://stackoverflow.com/questions/9894961/strange-warnings-from-the-linker-ld
https://developer.apple.com/library/archive/documentation/DeveloperTools/Conceptual/CppRuntimeEnv/Articles/SymbolVisibility.html#//apple_ref/doc/uid/TP40001670-98807

### CODE REVIEW

**Comment by battlmonstr** [comment]
@AndreaLanfranchi please help to check if it works on Windows

File: node/CMakeLists.txt [hunk 1]
```diff
@@ -22,7 +22,7 @@ find_package(gRPC CONFIG REQUIRED)
 get_filename_component(SILKWORM_MAIN_DIR ../ ABSOLUTE)
 
 file(GLOB_RECURSE SILKWORM_NODE_SRC CONFIGURE_DEPENDS "*.cpp" "*.hpp" "*.c" "*.h" "*.cc")
-list(FILTER SILKWORM_NODE_SRC EXCLUDE REGEX "_test\.cpp$")
+list(FILTER SILKWORM_NODE_SRC EXCLUDE REGEX "_test\\.cpp$")
```
Comment (yperbasis) [inline node/CMakeLists.txt:25]: Let's make the same change in `core/CMakeLists.txt`

File: cmake/Hunter/config.cmake [hunk 2]
```diff
@@ -32,6 +40,11 @@ hunter_config(
 hunter_config(
   Protobuf
   VERSION 3.14.0-4a09d77-p0
+  CMAKE_ARGS
```
Comment (yperbasis) [inline cmake/Hunter/config.cmake:43]: But aren't these cmake policies passed from our toolchain `cmake/toolchain/base.cmake` ?

Comment (battlmonstr) [reply cmake/Hunter/config.cmake:43]: Oh, I didn't know we already set them globally.

I see these variable values from `base.cmake` in config.cmake if I print them with `message()`,
but I'm not sure that they are propagated to the Protobuf library.

Inside
~/.hunter/_Base/Download/Hunter/0.24.3/10738b5/Unpacked/cmake/projects/Protobuf/hunter.cmake
they call `hunter_cmake_args()` ([declared here](https://github.com/ruslo/hunter/blob/master/cmake/modules/hunter_get_cmake_args.cmake)), but it is not clear to me whether it merges with the parent scope or overrides it.

I don't know how to verify it other than see if you get these link warnings before and after this patch.
Do you have them on master and on this branch?

Comment (yperbasis) [reply cmake/Hunter/config.cmake:43]: Hmm, I do get linker warning w/o these explicit `CMAKE_ARGS` for protobuf. Strange, how come Hunter doesn't propagate cmake settings from the toolchain 🤷‍♂️

**Pull request merged**